src/app/results/results.component.html

```diff
@@ -75,7 +75,7 @@
                 <ul class="pagination" id="pag-bar">
                     <li class="page-item"><span class="page-link" href="#" (click)="decPresentPage()">Previous</span></li>
                     <li class="page-item" *ngFor="let num of getNumber(maxPage)"><span class="page-link" *ngIf="presentPage>=4 && num<=noOfPages" [class.active_page]="getStyle(presentPage-3+num)"
-                            (click)="getPresentPage(presentPage-3+num)" href="#">{{presentPage-3+num}}</span>
+                            (click)="getPresentPage(presentPage-4+num)" href="#">{{presentPage-3+num}}</span>
```

Comment:
no listen when we send the getPresentPage(6) It should return me the 6th page. this is easier to understand what the function is doing.
in your case to return to 6th page you are passing input to the function as 5
one way you could modify this is by changing the getPresentPage function from
```
getPresentPage(N) {
    this.presentPage = N;
    this.searchdata.start = (this.presentPage) * this.searchdata.rows;
    this.route.navigate(['/search'], { queryParams: this.searchdata });
  }

```
to
```
getPresentPage(N) {
    this.presentPage = N;
    this.searchdata.start = (this.presentPage-1) * this.searchdata.rows;
    this.route.navigate(['/search'], { queryParams: this.searchdata });
  }
```
subtracting -1 in the `this.searchdata.start` line